### PR TITLE
Draft POC for tidyselect summarizers

### PR DESCRIPTION
**Pull request by DavisVaughan**

### CODE REVIEW

**Comment by DavisVaughan** [comment]
@EmilHvitfeldt okay this is ready for you to take a look at and pull pieces into #322 

It should actually be in a state where it passes CRAN checks, so it will mostly be a job of carefully copy/pasting into your current PR on top of the work you have done

If you can work out the merge conflicts, you could also rebase this PR onto that one and merge this one straight in, but that might be a little tricky.

Probably also needs a NEWS bullet, something like:

```r
* yardstick metric functions now use a pure tidyselect interface for `truth`, `estimate`, and the `...` of class probability metrics (#322).
```

File: tests/testthat/_snaps/template.md [hunk 1]
```diff
@@ -84,35 +80,30 @@
...
+        truth = obs, VF:L, obviouslywrong = TRUE)
     Condition
-      Error in `prob_metric_summarizer()`:
-      ! `...` must be empty.
-      x Problematic argument:
-      * obviouslywrong = TRUE
+      Error:
+      ! Must subset columns with a valid subscript vector.
+      x Subscript `TRUE` has the wrong type `logical`.
+      i It must be numeric or character.
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/template.md:108]: Weird case where `obviouslywrong = TRUE` actually gets passed through the `...` to the tidyselect function, but I'm not worried about this

File: tests/testthat/test-template.R [hunk 2]
```diff
@@ -474,7 +475,7 @@ test_that("prob_metric_summarizer()'s na_rm argument work", {
     fn = roc_auc_vec,
     data = hpc_f1_na,
     truth = obs,
-    estimate = matrix(data = c(VF = VF, F = F, M = M, L = L), ncol = 4L),
+    VF:L,
```
Comment (DavisVaughan) [inline tests/testthat/test-template.R:478]: New interface makes for easier testing for sure

**Comment by EmilHvitfeldt** [comment]
Thanks for all the work! close with https://github.com/tidymodels/yardstick/pull/322

**Pull request closed without merging**